src/gsi/read_satwnd.f90

```diff
@@ -1089,7 +1180,9 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
                 if(itype==251 )  then;  c_prvstg='GOESR' ; c_sprvstg='VIS'  ; endif
               endif
 ! Extra block for GOES-R winds: End
-
+           else ! wind is not recognised and itype is not assigned
```

Comment:
This else block catches the situation when a subset read from satwnd does not have a corresponding subset section of code in read_satwnd.f90.  It does not catch the case in which a match is found but itype is NOT set for other reasons.   An example of this was subset NC005091.   itype remained -1 because the wrong satid range was checked.   itype-1 caused the code to fail when run with array bounds checking on.  If itype remains-1 after checking the subset, something is wrong, right?   If "yes", what do we want to happen in this situation?

Reply:
I see your point. In that case I will add an additional:

if (itype==-1) then cycle loop_readsb   

after the two major if-then structures.